nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/NonDexMojo.java

```diff
@@ -59,33 +59,33 @@ a copy of this software and associated documentation files (the
 public class NonDexMojo extends AbstractNonDexMojo {
 
     private List<NonDexSurefireExecution> executions = new LinkedList<>();
+    private MojoExecutionException allExceptions = null;
 
     @Override
     public void execute() throws MojoExecutionException, MojoFailureException {
         super.execute();
         Logger.getGlobal().log(Level.INFO, "The original argline is: " + this.originalArgLine);
-        MojoExecutionException allExceptions = null;
         CleanSurefireExecution cleanExec = new CleanSurefireExecution(
                 this.surefire, this.originalArgLine, this.mavenProject,
                     this.mavenSession, this.pluginManager,
                     Paths.get(this.baseDir.getAbsolutePath(), ConfigurationDefaults.DEFAULT_NONDEX_DIR).toString());
 
         // If we add clean exceptions to allExceptions then the build fails if anything fails without nondex.
         // Everything in nondex-test is expected to fail without nondex so we throw away the result here.
-        this.executeSurefireExecution(allExceptions, cleanExec);
-
-        for (int i = 0; i < this.numRuns; i++) {
-            NonDexSurefireExecution execution =
-                    new NonDexSurefireExecution(this.mode, this.computeIthSeed(i),
-                            Pattern.compile(this.filter), this.start, this.end,
-                            Paths.get(this.baseDir.getAbsolutePath(), ConfigurationDefaults.DEFAULT_NONDEX_DIR).toString(),
-                            Paths.get(this.baseDir.getAbsolutePath(), ConfigurationDefaults.DEFAULT_NONDEX_JAR_DIR)
-                                .toString(),
-                            this.surefire, this.originalArgLine, this.mavenProject,
-                            this.mavenSession, this.pluginManager);
-            this.executions.add(execution);
-            allExceptions = this.executeSurefireExecution(allExceptions, execution);
-            this.writeCurrentRunInfo(execution);
+        if (this.mode.toString().equals("SYSTEMATIC")) {
+            // In certain environments calls made to nextInt() from NonDex itself start at 59, if the user wants to change
+            // this variable they can use the DnondexStart argument
+            if (this.start == 0) {
+                this.start = 59;
+            }
+            do {
+                executeHelper(1);
+            } while (Files.exists(Paths.get(ConfigurationDefaults.DEFAULT_SYSTEMATIC_LOG)));
```

Comment:
Oh, I see why you are removing it, because that's your stopping condition... You'll need to change this to communicate in some other way that exploration is finished. For example, by creating a file named explorationDone in which you store the results. On second thought, you could even simply rename the systematicLog file to systematicDone; this way your check will still work here, but you can still write the assertion I mentioned before and the log is still available for debugging.